yaksh/cpp_stdio_evaluator.py

```diff
@@ -0,0 +1,105 @@
+            self.ref_output_path
+            )
+        self.compiled_user_answer = self._run_command(self.compile_command,
+                                                      shell=True,
+                                                      stdin=subprocess.PIPE,
+                                                      stdout=subprocess.PIPE,
+                                                      stderr=subprocess.PIPE
+                                                      )
+        self.compiled_test_code = self._run_command(self.compile_main,
+                                                    shell=True,
+                                                    stdin=subprocess.PIPE,
+                                                    stdout=subprocess.PIPE,
+                                                    stderr=subprocess.PIPE
+                                                    )
+        return self.compiled_user_answer, self.compiled_test_code
+
+    def check_code(self, user_answer, expected_input, expected_output):
+        success = False
+        proc, stdnt_out, stdnt_stderr = self.compiled_user_answer
+        stdnt_stderr = self._remove_null_substitute_char(stdnt_stderr)
+        if stdnt_stderr == '':
+            proc, main_out, main_err = self.compiled_test_code
+            main_err = self._remove_null_substitute_char(main_err)
+            if main_err == '':
+                proc = subprocess.Popen("./executable",
+                                        shell=True,
+                                        stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE
+                                        )
+                evaluator = Evaluator()
+                success, err = evaluator.evaluate(user_answer, proc,
+                                                  expected_input,
+                                                  expected_output
+                                                  )
+                os.remove(self.ref_output_path)
+            else:
+                err = "Error:"
+                try:
+                    error_lines = main_err.splitlines()
+                    for e in error_lines:
+                        if ':' in e:
+                            err = err + "\n" + e.split(":", 1)[1]
+                        else:
+                            err = err + "\n" + e
+                except:
+                        err = err + "\n" + main_err
+            os.remove(self.user_output_path)
+        else:
+            err = "Compilation Error:"
+            try:
+                error_lines = stdnt_stderr.splitlines()
+                for e in error_lines:
+                    if ':' in e:
+                        err = err + "\n" + e.split(":", 1)[1]
+                    else:
+                        err = err + "\n" + e
+            except:
+                err = err + "\n" + stdnt_stderr
```

Comment:
We can have a method for error handling, which will produce error in a proper format. This can be done later. For now I am raising an issue.